### PR TITLE
Pose exp comp

### PR DESCRIPTION
**Pull request by greenblitz4590**
sorry );

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/RobotMap.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public static class General {
 		public final static double minVoltageBattery = 11.97;
 		public final static double VOLTAGE_COMP_VAL = 11.5;
 		public final static double RAMP_RATE_VAL = 0.4;
-		
+		public final static double ITERATION_DT	= 0.021;
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/RobotMap.java:31]: Broder - why not 0.02?

File: src/main/java/edu/greenblitz/tobyDetermined/RobotMap.java [hunk 2]
```diff
@@ -173,7 +173,7 @@ public static class KazaSwerve {
 			public static final PIDObject linPID = new PIDObject().withKp(0.0003).withMaxPower(0.5);
 			public static final GBSparkMax.SparkMaxConfObject baseLinConfObj = new GBSparkMax.SparkMaxConfObject().withIdleMode(CANSparkMax.IdleMode.kBrake).withCurrentLimit(40).withRampRate(RobotMap.General.RAMP_RATE_VAL).withVoltageComp(RobotMap.General.VOLTAGE_COMP_VAL).withPID(linPID).withPositionConversionFactor(linTicksToMeters).withVelocityConversionFactor(linTicksToMetersPerSecond);
 			public static final double angleTicksToRadians = RobotMap.General.Motors.SPARKMAX_TICKS_PER_RADIAN / ANG_GEAR_RATIO;
-			public static final PIDObject angPID = new PIDObject().withKp(0.5).withMaxPower(1.0);
+			public static final PIDObject angPID = new PIDObject().withKp(0.4).withMaxPower(1.0);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/RobotMap.java:176]: y?

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java [hunk 3]
```diff
@@ -16,7 +16,6 @@
 public class Limelight extends GBSubsystem {
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/Limelight.java:16]: broder - check if this is right

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java [hunk 4]
```diff
@@ -48,6 +53,20 @@ public SwerveChassis() {
...
         SmartDashboard.putData("field", getField());
         field.getObject("apriltag").setPose(RobotMap.Vision.apriltagLocation.toPose2d());
+
+        moduleUpdater = new Thread(() -> {
+            while(true){
+                moveByChassisSpeeds(wantedSpeeds);
+                SmartDashboard.putNumber("cycle time", (Timer.getFPGATimestamp() - lastUpdate) * 1000);
+                lastUpdate = Timer.getFPGATimestamp();
+                try {
+                    Thread.sleep(20);
```
Comment (greenblitz4590) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/swerve/SwerveChassis.java:63]: Use the const you set

**Comment by greenblitz4590** [comment]
dont merge without consulting asaf

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToPos.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
+package edu.greenblitz.tobyDetermined.commands.swerve;
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToPos.java:1]: No. We wont have MoveToPos and MoveToPose

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToPos.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+public class MoveToPos extends SwerveCommand {
+    /**
+     * get to a given pose with 3 pidControllers - x, y, rotational<3
+     */
+    private ProfiledPIDController xController;
+    private ProfiledPIDController yController;
+    private ProfiledPIDController rotationController;
+    private Pose2d pos;
+
+    public static final double rotKp = 2;
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToPos.java:21]: Why all of those not in robot map?

File: src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToPos.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        xController.reset(swerve.getRobotPose().getX(), swerve.getChassisSpeeds().vxMetersPerSecond);
+        yController.reset(swerve.getRobotPose().getY(), swerve.getChassisSpeeds().vyMetersPerSecond);
+        rotationController.reset(swerve.getRobotPose().getRotation().getRadians(), swerve.getChassisSpeeds().omegaRadiansPerSecond);
+        xController.setGoal(new TrapezoidProfile.State(pos.getX(), 0));
+        yController.setGoal(new TrapezoidProfile.State(pos.getY(), 0));
+        rotationController.setGoal(new TrapezoidProfile.State(pos.getRotation().getRadians(), 0));
+        
+    }
+    
+    public void debugDashboard(double xCalc, double yCalc, double rotationCalc){
```
Comment (roygold8) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/swerve/MoveToPos.java:56]: probably should remove

**Review by roygold8 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**